pysteps/io/importers.py

```diff
@@ -131,6 +133,230 @@
 except ImportError:
     PYPROJ_IMPORTED = False
 
+try:
+    import pygrib
+
+    PYGRIB_IMPORTED = True
+except ImportError:
+    PYGRIB_IMPORTED = False
+
+
+def _check_coords_range(selected_range, coordinate, full_range):
+    """Check that the coordinates range arguments follow the expected pattern in the **import_mrms** function."""
+
+    if selected_range is None:
+        return sorted(full_range)
+
+    if not isinstance(selected_range, (list, tuple)):
+
+        if len(selected_range) != 2:
+            raise ValueError(f"The {coordinate} range must be None or a two-element tuple or list")
+
+        selected_range = list(selected_range)  # Make mutable
+
+        for i in range(2):
+            if selected_range[i] is None:
+                selected_range[i] = full_range
+
+        selected_range.sort()
+
+    return tuple(selected_range)
+
+
+def import_mrms(filename, fillna=np.nan, lat_range=None, lon_range=None,
+                dtype='float32', block_size=4, **kwargs):
+    """
+    Importer for NSSL's Multi-Radar/Multi-Sensor System
+    ([MRMS](https://www.nssl.noaa.gov/projects/mrms/)) rainrate product
+    (grib format).
+
+    The rainrate values are expressed in mm/h, and the dimensions of the data
+    array are [latitude, longitude]. The first grid point (0,0) corresponds to
+    the upper left corner of the domain, while (last i, last j) denote the
+    lower right corner.
+
+    Due to the large size of the dataset (3500 x 7000), a float32 type is used
+    by default to reduce the memory footprint. However, be aware that when this
+    array is passed to a pystep function, it may be converted to double
+    precision, doubling the memory footprint.
+    To change the precision of the data, use the *dtype* keyword.
+
+    Also, by default, the original data is downscaled by 4
+    (that gives a 4 km grid spacing).
+    In case that the original grid spacing is needed, use `block_size=1`.
+    But be aware that a single composite in double precipitation will
+    require 186 Mb of memory.
+
+    Finally, the precipitation data can only be extracted sub region of the
+    full domain using the `lat_range` and `lon_range` keywords.
+    By default, the entire domain is returned.
+
+    Parameters
+    ----------
+
+    filename : str
+        Name of the file to import.
+
+    fillna : float or np.nan
+        Value used to represent the missing data ("No Coverage").
+        By default, np.nan is used.
+
+    dtype : str
+        Data-type to which the array is cast.
+        Valid values:  "float32", "float64", "single", and "double".
+
+    lat_range : tuple, list, or None
+        Latitude range (min_lat, max_lat) of the data to be retrieved.
+        By default, the entire latitude range is retrieved.
+
+    lon_range : tuple, list, or None
+        Longitude range (min_lon, max_lon) of the data to be retrieved.
+        By default, the entire longitude range is retrieved.
```

Comment:
The `lat_range` and `lon_range` arguments have somewhat the same functionality as the `extent` argument in the `import_saf_crri` method. If possible, I suggest using the same approach for both methods to provide some consistency among the importers.

Reply:
Good point. I made the change but the way the `extent` is used in the MRMS importer is different from the `import_saf_crri` function. `import_mrms` uses the extent in degrees while `import_saf_crri` uses the data coordinates units (km). 

The `extent` keyword can be also applied to the other importers, allowing the user to select the units of the extent (km or degrees). This is a good candidate for a separate PR (including the data type casting).

Reply:
Excellent, thanks!

I don't see it as a different use of `extent` actually: both use the (left, right, bottom, top) in *data coordinates* to specify bounding box for a regular grid. Whether this is in km or degrees doesn't really matter.

Yes, good idea, extent could be implemented for all importers.

Reply:
You are right. The actual difference is in the units. We can add another keyword to specify this, like extend_units='latlon' or 'km'. 

I will add it to this PR since it will become handy if we want this functionality in all the importers.

Reply:
I wonder whether this is actually needed, since extent is to be specified in the same units of the dataset. In most use cases, one would want to subset the field using the same coordinates. Or am I missing something here?

Reply:
So far, the importers retrieve the data coordinates in "km" or "m" with respect to the origin. However, we can also use the lat/lon associated with the grid points to define the extent of the data that we want to retrieve.  The domain of the MRMS quite large (the contiguous US), if you want to focus on a single state, you simply specify the lat-lon bounding box. On the other hand, to use the data coordinates, you first need to find the bounding box coordinates measured from the origin, which will require an additional step. 

We can have as default behavior the coordinates specified in the same units of the dataset, and leave the lan/lon units as a particular use-case. What do you think?

Reply:
It worth mentioning that the `extend units` are unnecessary for the MRMS importer, since the data coordinates [x, y] are the same as the [lon, lat] (a regular lat/lon projection is used).

However, it may be useful for the other importers where a projection that transforms the lat/lon coordinates to x/y native map projection coordinates (in meters).

Reply:
I see your point, but the `extent` argument as it is defined now (i.e. [left, right, bottom, top] coordinates) can only work with a regular grid. To be able to accept bounding boxes in lat lon for any projection, we would need 4 pairs of coordinates (the 4 corners of the bounding box). For the MRMS data, the regular grid is defined using a geographical projection in lat lon, so the extent can be specified in degrees.

I suggest keeping the same approach as it is now, which closely mimic the behavior of the `extent` argument in the pyplot.imshow() method. In brief, the extent is specified in whatever coordinates a given dataset is based on. This will ensure that the extent is always with respect to a regular grid.

Reply:
> To be able to accept bounding boxes in lat lon for any projection, we would need 4 pairs of coordinates (the 4 corners of the bounding box). 
I see the possible issue now. You are right, let's keep it as it is defined.